Fix possible null value being passed to parse_str()

SebKay opened the pull request:
This fixes a PHP deprecation warning in PHP 8.0+ where a potentially `null` value can be passed to `parse_str()`.

### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?


### Changes proposed in this Pull Request:

- [x] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

SebKay committed:
Fix possible null value being passed to parse_str()

SebKay committed:
Fix possible null value being passed to parse_str()

zhongruige committed:
Add an NPM script to run the Playwright API Core Tests (#35283)

* Added script for running Playwright API Core Tests

* Added changelog

mdperez86 committed:
Remove some placeholder values (#35267)

nigeljamesstevenson committed:
add webhooks api-core-tests (#35292)

zhongruige committed:
Update Playwright API Core Tests readme (#35303)

* Update api core tests readme

* Added changelog

* Minor spacing tweak

peterfabian committed:
Update states.php -> added Senegal Regions (#35199)

* Update states.php -> added Senegal Regions

* Update to CLDR.

* Changelog.

Co-authored-by: Papa Amadou Korka Sow <32936698+Sowgenius@users.noreply.github.com>

mattsherman committed:
Fix DateTimePickerControl suffix style (#35256)

mdperez86 committed:
Improve the communication around required and optional (#35266)

* Improve the communication around required and optional

* Add comments suggestions

mdperez86 committed:
Change the product info section title to Product Details (#35255)

* Change the product info section title to Product Details

* Add comment suggestions

mattsherman committed:
Support PHP style date format specifiers in DateTimePickerControl (#35285)

* Use @wordpress/date to format for display in DateTimePickerControl
* Update default formats to PHP style
* Update tests to use @wordpress/date formatting
* Update CustomDateTimeFormat story to use PHP style formatting

mdperez86 committed:
Update font size and spacing in the tooltip component (#35265)

mdperez86 committed:
Increase the spacing between the shipping box illustration and the dimensions fields (#35259)

mdperez86 committed:
Fix the display of letter descenders in the shipping class dropdown menu (#35258)

* Fix the display of letter descenders in the shipping class dropdown menu

* Add comment suggestions

jonathansadowski committed:
Fix reference to HEAD to work with community PRs (#35315)

chihsuan commented on plugins/woocommerce/src/Admin/PageController.php:
```diff
@@ -124,7 +124,7 @@ public function determine_current_page() {
 		}
 
-		$current_query = wp_parse_url( $current_url, PHP_URL_QUERY );
+		$current_query = wp_parse_url( $current_url, PHP_URL_QUERY ) ?: '';
```
@SebKay Thanks for fixing this! I think it's safer to use `??` here since `0` would be considered `false`. What do you think? 🙂 

```suggestion
		$current_query = wp_parse_url( $current_url, PHP_URL_QUERY ) ?? '';
```

chihsuan commented:
@SebKay Could you add a changelog via `pnpm --filter=woocommerce changelog add`? Thanks.

chihsuan committed:
Fix changelog command in PR template (#35326)

alexmigf committed:
Uniform order customer search and display code (#35244)

Co-authored-by: Néstor Soriano <konamiman@konamiman.com>

iamdharmesh committed:
Fixed "Unsupported operand types" error. (#34327)

Zaglov committed:
Cat dashboard loading time (#34292)

Co-authored-by: Néstor Soriano <konamiman@konamiman.com>

nigeljamesstevenson committed:
Add/a2 p create settings crud api core tests (#35253)

* api-core-tests settings tests

* api-core-tests settings tests

* api-core-tests settings tests

* update settings test to ensure concurrent tests do not fail

jonathansadowski committed:
Fix line endings in change file from 34292 (#35334)

* Fix line endings in change file from 34292

* Correctly update change file

alopezari committed:
Updated COT plugin used to set up the local environment with COT enabled. (#34990)

* Updated COT plugin used to set up the local environment with COT enabled.

* Enable HPOS in the e2e environment using env var (#35057)

* Updated COT plugin used to set up the local environment with COT enabled.

* Used env var to enable HPOS

* Disable HPOS for performance tests

* Updated permissions

* Set ENABLE_HPOS to 1

Co-authored-by: Alex López <alex.lopez@automattic.com>

Co-authored-by: Jamel Noel Reid <MrJnrman@users.noreply.github.com>

joshuatf committed:
Add summary field to new product experience (#35201)

* Add summary to new product page experience

* Add changelog entry

lanej0 committed:
Skip flakey settings API test. (#35338)

Co-authored-by: Jon Lane <jon.lane@automattic.com>

MrJnrman committed:
Revert setting up permalinks in PW global setup (#35337)

* Revert permalinks setup

* Add changelog

Co-authored-by: Jon Lane <jon.lane@automattic.com>

opr committed:
Update `@woocommerce/extend-checkout-block` to include an example of a forced inner block (#33833)

* Update main integration file to register newsletter block scripts/styles

* Add files for example newsletter block

* Use only one internal dependencies comment

* Include newsletter block in index

* Move entry to correct position in webpack

* Rename functions to register scripts and styles

* Add changelog

* Remove block registration code

* Register slug as a block category to prevent warnings in console

* Add comment explaining the purpose of the function

* Bring templates up to date with latest changes to WC Blocks

The way the validation store selectors work was updated, so this template will use the new selectors.

* Ensure error message shows when box is not checked

* Ensure checkbox is targeted with CSS in editor

SebKay committed:
Update plugins/woocommerce/src/Admin/PageController.php

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

SebKay committed:
Update plugins/woocommerce/src/Admin/PageController.php

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

SebKay committed:
Add changelog entry

SebKay committed:
Add changelog entry

SebKay commented:
> @SebKay Could you add a changelog via `pnpm --filter=woocommerce changelog add`? Thanks.

Done!

louwie17 committed:
Add name to select control popover slots (#35353)

* Add name to SelectControlMenuSlot

* Add changelogs

* Update changelog

* Fix typo in changelog

roykho committed:
Add line number reference links to release posts for hooks (#35354)

vedanshujain committed:
Remove address indexes from list of internal keys. (#35192)

* Override filter_meta_data method, since it should be a no-op anyway.

* Add changelog.

* Not include address indexes from filtered data.

* Applied coding standards.

chihsuan commented:
Hey @SebKay A required check ("Run code sniff") is failing due to https://github.com/woocommerce/woocommerce/issues/35369. We'll merge this once the issue is fixed. Thanks!

chihsuan committed:
Migrate @woocommerce/data report store to TS (#35048)

* Migrate reports store to TS

* Add changelog

* Update reports types

roykho committed:
Fix release post not parsing certain versions correctly (#35363)

mattsherman committed:
DateTimePickerControl: Allow time to be set to beginning or end of day when in date-only mode (#35296)

peterfabian committed:
TT3 compatibility (#35306)

jorgeatorres committed:
[HPOS] Allow line breaks in order notes (admin-side) (#35366)

Allow line breaks in order notes (admin-side).

* Fix code violations

barryhughes committed:
Guard against cases where `get_current_screen()` is undefined. (#35371)

This may happen if code invokes the `all_plugins` filter during non-admin requests, for example.

louwie17 committed:
Add product description title in old product editor (#35154)

* Add product description title to classic product editor

* Remove bottom margin as its already added by the parent element

* Add changelog

* Update css and name, also fix some lint errors

* Fix spacing

github-actions[bot] committed:
Delete changelog files based on PR 35306 (#35391)

Delete changelog files for 35306

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

louwie17 committed:
Update shipping label banner add meta boxes function (#35212)

* Add check to see if we are on a shop_order page.

* Update deprecated function call

* Add changelog

* Fix yoda condition

github-actions[bot] committed:
Delete changelog files based on PR 35371 (#35400)

Delete changelog files for 35371

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

vedanshujain committed:
Check order type is set before returning to prevent notice. (#35349)

* Check order type is set before returning to prevent notice.

* Applied code standards.

* Remove type declaration since its not consistent with CPT datastore.

* Switch to a yoda condition (satisfy required linting check).

Co-authored-by: barryhughes <3594411+barryhughes@users.noreply.github.com>

vedanshujain committed:
Also support syncing for HPOS with stats table. (#35118)

* Also support syncing for HPOS with stats table.

* Add changelog.

* Fixup.

* More fixup.

* test commit by disabling importing admin test orders.

* Update get_items to work with HPOS.

* Modify tests to assert against invalid result.

* test commit for ci.

* Remove seperate test as its quite slow.

* Applied coding standards.

* Coding standards, part 2.

Konamiman committed:
Exclude inactive plugins from the incompatible plugins view (#35333)

* Changes in the plugin-feature compatibility warnings:

- Show only active plugins in the "Incompatible with..." plugins view
- Show an empty page when the view is loaded and no Incompatible
  plugins are active (instead of listing all the existing plugins)

* Add changelog file

psealock committed:
PHPCS: Lint only changes (#35407)

Add `sirbrillig/phpcs-changed` to run PHPCS only on changes, not entire files

jonathansadowski committed:
Update/blocks 8.7.5 (#35428)

* Update blocks to 8.7.5

* Add changelog file

nigeljamesstevenson committed:
add/a2p data api-core-tests (#35347)

* add data api-core-tests

* add newline to see if this resolves changelog issue

* add newline to see if this resolves changelog issue

roykho committed:
Restore previously installed pnpm version (#35389)

nigeljamesstevenson committed:
add/a2p reports api-core-tests (#35388)

add reports api-core-tests

louwie17 committed:
Fix/form ts error (#35394)

* Fix forwardRef type in the Form component

* Add changelog

* Revert type changes

barryhughes committed:
Ensure the HPOS/COT order status correctly tracks the CPT order status (#35402)

Ensure the HPOS/COT order status correctly tracks the CPT order status (during manual order creation).

* Linting.
* Address coding standards (no need to prepare the query when there are no placeholders).
* Woops! Restore use of `$wpdb->prepare()`, add phpcs:ignore rule.

joshuatf committed:
Add inventory advanced section (#35164)

* Add sold individually inventory option

* Add backorder purchase options

* Add margin around collapsible content areas

* Add changelog entry

* Add tests around inventory section

* Fix up checkbox props after rebase

* Check for disabled track quantity toggle

* Update manage stock test

* Fix nested radio control label margin

joshuatf committed:
Fix up rich text editor initial selection and add blocks (#35286)

* Fix double click toolbar behavior

* Fix initial block selection on editor load

* Add placeholder option to RichTextEditor

* Add image and video blocks

* Set toolbar height

* Allow inserter to be shown

* Allow media uploads in rich text editor

* Add changelog entries

* Fix media upload

* Check for existence of selected blocks before checking length

* Pass blocks to avoid race in detecting initially empty blocks

github-actions[bot] committed:
Delete changelog files based on PR 35428 (#35430)

Delete changelog files for 35428

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 35118 (#35433)

Delete changelog files for 35118

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 35366 (#35435)

Delete changelog files for 35366

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 35402 (#35437)

Delete changelog files for 35402

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 35333 (#35439)

Delete changelog files for 35333

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

jonathansadowski committed:
Remove change file for 35349 (#35443)

louwie17 committed:
Add create attribute term modal (#35131)

* Add create attribute term modal

* Add back filter missed during rebase

* Add changelog

* Fix lint error

* Address some feedback from PR review

* Prevent first modal from closing if closing the second modal when clicking outside

nigeljamesstevenson committed:
add/a2p product variations api-core-tests (#35355)

* add product variations api-core-tests

* add product variations api-core-tests

* comments updates

nigeljamesstevenson committed:
update/a2p update shipping api-core-tests (#35332)

* update shipping api-core-tests

* merge in trunk to see if this resolves changelog issue

* comments updates

joshuatf committed:
Remove insight first product and payment note (#35309)

* Remove InsightFirstProductAndPayment note

* Add changelog entry

* Remove obsolete note

* Fix yoda condition

barryhughes committed:
Add filter hook `handle_bulk_actions-woocommerce_page_wc-orders` (#35442)

Add filter hook `handle_bulk_actions-woocommerce_page_wc-orders` to the HPOS admin list table.

This is a duplicate of core WP hook `handle_bulk_actions-<SCREEN_ID>` and allows for custom bulk actions to be handled in the context of the admin list table for HPOS orders.

kkmuffme committed:
Fix missing use FQCN for WP_Error (#35305)

* add use WP_Error

phpdoc was not using FQCN for WP_Error, however use is preferred to FQCN anyway
Fix: https://github.com/woocommerce/woocommerce/issues/35304

* Revert "add use WP_Error"

This reverts commit ff62deb10020689f96722e10c5c86669a1454125.

* use FQCN in PHPDoc when using FQCN in code

* add changelog

kkmuffme committed:
Fix class name for class FirstDownlaodableProduct (#35383)

Spelling

Fix https://github.com/woocommerce/woocommerce/issues/35307

chihsuan commented:
Hey @SebKay Could you please rebase your branch? We've fixed the issue. Will merge this once all required checks pass. Thank you!

chihsuan committed:
Fix invalid `wcadmin_install_plugin_error` event props (#35411)

* Fix invalid wcadmin_install_plugin_error props

* Add changelog

* Update docblock comments to fix lint

* Fix lint

* Update prop name

SebKay committed:
Merge branch 'trunk' of https://github.com/SebKay/woocommerce into trunk

chihsuan commented:
@SebKay It looks like it accidentally brings in all other unrelated commits here due to the merge. Could you remove them?

SebKay commented:
> @SebKay It looks like it accidentally brings in all other unrelated commits here due to the merge. Could you remove them?
> 
> https://github.com/woocommerce/woocommerce/pull/35289/commits

Ah, sorry about that!

I'm going to be honest, I have no idea how to do that now the commits are already pushed :-/

chihsuan commented:
> I'm going to be honest, I have no idea how to do that now the commits are already pushed :-/

No worries, you can rebase your branch using the following commands:

```bash
git remote add upstream https://github.com/woocommerce/woocommerce.git
git fetch upstream
git rebase upstream/trunk
# Resolve conflicts in plugins/woocommerce/src/Admin/PageController.php
git add plugins/woocommerce/src/Admin/PageController.php
git rebase --continue
git push -f
```

chihsuan commented:
> No worries, you can rebase your branch using the following commands:

Hey @SebKay Did you try that? Let me know if you get stuck. Thanks!

chihsuan commented:
@SebKay I'm closing this for now. Feel free to reopen it. Thanks for your contribution!
